src/toolchain.ml

```diff
@@ -110,6 +110,14 @@ module Package_manager = struct
     let t = Settings.create ~scope:Workspace ~key:"sandbox" ~of_json ~to_json
   end
 
+  let equal pm1 pm2 =
```

Comment:
This is not a general purpose equality function, so it's quite confusing to call it that.

Reply:
Didn't quite get what you mean. Isn't `M.equal: M.t -> M.t -> bool` a common function defined for modules?

Reply:
`equal` is the common name for structural equality (or something that is morally equivalent). What you're implementing here can give us equality for:

```ocaml
let x = Opam (Opam.make (), Named "foo"))
let y = Opam (Opam.make (), Named "bar"))
```

Even if the `Opam.t` returned isn't equal.

There's nothing wrong with having a function like that, but it shouldn't be called `equal`.